crates/exec-wasmtime/src/loader/configured/platform.rs

```diff
@@ -65,29 +68,83 @@ impl Platform {
         }
 
         match (rax, rdx) {
-            (ENOSYS | EPERM, ..) => Ok(Self(Technology::Kvm, 0)),
+            (ENOSYS | EPERM, ..) => Ok((Technology::Kvm, 0)),
             (n, ..) if n < 0 => Err(std::io::Error::from_raw_os_error(-n as i32)),
             (n, t) => match t {
-                0 => Ok(Self(Technology::Kvm, n as _)),
-                1 => Ok(Self(Technology::Snp, n as _)),
-                2 => Ok(Self(Technology::Sgx, n as _)),
+                0 => Ok((Technology::Kvm, n as _)),
+                1 => Ok((Technology::Snp, n as _)),
+                2 => Ok((Technology::Sgx, n as _)),
                 _ => Err(ErrorKind::Other.into()),
             },
         }
     }
 
+    #[cfg(not(all(target_os = "linux", target_arch = "x86_64")))]
+    fn get_key(mut buf: Option<&mut [u8]>) -> Result<usize> {
+        Ok(0)
+    }
+
+    #[cfg(all(target_os = "linux", target_arch = "x86_64"))]
+    fn get_key(mut buf: Option<&mut [u8]>) -> Result<usize> {
+        use sallyport::item::enarxcall::SYS_GETKEY;
+        use std::arch::asm;
+        use std::ptr::null_mut;
+
+        const ENOSYS: isize = -(libc::ENOSYS as isize);
+        const EPERM: isize = -(libc::EPERM as isize);
+
+        let mut rax: isize;
+
+        unsafe {
+            asm!(
+            "syscall",
+            lateout("rax") rax,
+            in("rax") SYS_GETKEY,
+            in("rdi") buf.as_mut().map(|x| x.as_mut_ptr()).unwrap_or_else(null_mut),
+            in("rsi") buf.map(|x| x.len()).unwrap_or_default(),
+            lateout("rcx") _, // clobbered
+            lateout("r11") _, // clobbered
+            )
+        }
+
+        match rax {
+            ENOSYS | EPERM => Ok(0),
```

Comment:
because if the backend is `nil` the Linux kernel tells us with ENOSYS that there is no such enarx syscall. If there is a syscall filter with seccomp  or similar (on `nil`) we would get EPERM.

Reply:
Anyway, on `nil` the return value is `0`.